ui/views/Authentication/ResetPasswordView.tsx

```diff
@@ -0,0 +1,168 @@
+import { ArrowBack } from '@mui/icons-material';
+import { Button, CircularProgress, TextField } from '@mui/material';
+import classnames from 'classnames';
+import { Formik, FormikErrors, FormikHelpers } from 'formik';
+import { useSnackbar } from 'notistack';
+import React, { Fragment, useEffect } from 'react';
+import { Link as RouterLink, useLocation, useNavigate } from 'react-router-dom';
+import Logo from 'assets';
+import useResetPassword from 'shared/authentication/actions/resetPassword';
+
+interface ResetPasswordValues {
+  password: string;
+  verifyPassword: string;
+}
+
+const initialValues: ResetPasswordValues = {
+  password: '',
+  verifyPassword: '',
+};
+
+export default function ResetPasswordView(): JSX.Element {
+  const { enqueueSnackbar } = useSnackbar();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const resetPassword = useResetPassword();
+
+  const search = location.search;
+  const query = new URLSearchParams(search);
+  const token = query.get('token');
+
+  useEffect(() => {
+    if (!token) {
+      navigate('/login');
+      enqueueSnackbar('You must get a password reset link to change your password.', {
+        variant: 'warning',
+        disableWindowBlurListener: true,
+      });
+    }
+  }, [token, enqueueSnackbar, navigate]);
+
+  function validateInput(values: ResetPasswordValues): FormikErrors<ResetPasswordValues> {
+    let errors: FormikErrors<ResetPasswordValues> = {};
+
+    if (values.password) {
```

Comment:
Should add something to trim the space with the password. Leading/trailing spaces are trimmed by the API anyway.